File: oit_fn/func.js
const fdk = require('@fnproject/fdk');
const fs = require('fs');
const tmp = require('tmp');
const im = require('imagemagick');
const { exec, spawn } = require('child_process');

fdk.handle((buffer, ctx) => {
	// let name = "World";
	// if(buffer.name) {
	// 	name = buffer.name;
	// }
	// return {'message': 'Hello ' + name};
	//exec('/function/oit-fi/sdk/demo/fisimple /function/oit-fi/sdk/samplefiles/adobe-acrobat.pdf', (err, stdout, stderr) => {
	try {
		const echo = spawn('ls', {shell: true});
		echo.stdout.on('data', (data)=> {
			return { 'message': `OUTPUT: ${data}` };
		});
		// exec('echo "Hello World"', {shell: true}, (error, stdout, stderr) => {
		// 	if (error) {
		// 		console.log(`error: ${error.message}`);
		// 		return;
		// 	}
		// 	if (stderr) {
		// 		console.log(`stderr: ${stderr}`);
		// 		return;
		// 	}
		// 	return { 'message': `OUTPUT: ${stdout}` };
		// });
	}
	catch(e) {
		return { 'message': 'Some Error: ' + e };
	}
	// return { 'message': 'Nothing is working again' };
})

File: oit_fn/func.py
import io
import json
import logging
from subprocess import check_output
import os

from fdk import response


def handler(ctx, data: io.BytesIO=None):
    name = "World"
    output = "Divyaksh"
    try:
        body = json.loads(data.getvalue())
        name = body.get("name")
    except (Exception, ValueError) as ex:
        logging.getLogger().info('error parsing json payload: ' + str(ex))

    # output = check_output('echo \"Hello World\"').decode('utf-8')
    # output = os.popen('echo \"Hello World\"').read().strip()
    # command = "/function/oit-fi/sdk/demo/fisimple /function/oit-fi/sdk/samplefiles/adobe-acrobat.pdf"
    
    output = os.popen("/function/oit-fi/sdk/demo/fisimple /function/oit-fi/sdk/samplefiles/adobe-acrobat.pdf").read().strip()

    logging.getLogger().info("Inside Python Hello World function")
    return response.Response(
        ctx, response_data=json.dumps(
            {"output": "{}".format(output)}),
        headers={"Content-Type": "application/json"}
    )
